src/Elcodi/Component/Geo/Validator/Constraints/CityExistsValidator.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/*
+ * This file is part of the Elcodi package.
+ *
+ * Copyright (c) 2014 Elcodi.com
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * Feel free to edit as you please, and have fun.
+ *
+ * @author Marc Morera <yuhu@mmoreram.com>
+ * @author Aldo Chiecchia <zimage@tiscali.it>
+ * @author Elcodi Team <tech@elcodi.com>
+ */
+
+namespace Elcodi\Component\Geo\Validator\Constraints;
+
+use Doctrine\ORM\EntityNotFoundException;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+use Elcodi\Component\Geo\Services\Interfaces\LocationProviderInterface;
+use Elcodi\Component\Geo\ValueObject\LocationData;
+
+/**
+ * Class CityExistsValidator
+ */
+class CityExistsValidator extends ConstraintValidator
+{
+    /**
+     * @var LocationProviderInterface
+     *
+     * A location provider
+     */
+    private $locationProvider;
```

Comment:
We are generally using `protected` accessors in Elcodi services to maximize  *open/closed*-ness